addon/helpers/hook.js

```diff
@@ -6,10 +6,38 @@ import returnWhenTesting from 'ember-hook/utils/return-when-testing';
 
 const { Helper } = Ember;
 
-export default Helper.extend({
-  compute(params, qualifiers = {}) {
-    const [hook] = params;
+/**
+ * Test if an object is empty `Ember.isEmpty` surprisingly doesn't do that
+ * @see {@link https://github.com/emberjs/ember.js/issues/5543}
+ * @see {@link https://github.com/emberjs/ember.js/blob/v2.10.0/packages/ember-metal/lib/is_empty.js}
+ *
+ * @param {Object} obj - the object to test
+ * @returns {Boolean} true if there are no keys on the object
+ */
+function isEmpty (obj) {
```

Comment:
The previous implementation of the hook helper didn't seem to do any type checking for the `hook` property itself, so I didn't think there needed to be any for the additional properties coming in now. If you'd like me to add type checking with exceptions thrown, I can certainly add that.